From 25425878243c5ca1ff21e6f479e585c60b943930 Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 30 Mar 2009 19:56:29 -0400
Subject: [PATCH] libsandbox: handle more *at functions

Add some more *at functions to the main checking code.

URL: http://bugs.gentoo.org/264320
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Reported-by: Harald van Dijk <truedfx@gentoo.org>
---
 libsandbox/libsandbox.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/libsandbox/libsandbox.c b/libsandbox/libsandbox.c
index 88248af..c3f0b55 100644
--- a/libsandbox/libsandbox.c
+++ b/libsandbox/libsandbox.c
@@ -681,15 +681,20 @@ static int check_access(sbcontext_t *sbcontext, int sb_nr, const char *func,
 	    sb_nr == SB_NR_CREAT       ||
 	    sb_nr == SB_NR_CREAT64     ||
 	    sb_nr == SB_NR_MKDIR       ||
+	    sb_nr == SB_NR_MKDIRAT     ||
 	    sb_nr == SB_NR_MKNOD       ||
 	    sb_nr == SB_NR_MKNODAT     ||
 	    sb_nr == SB_NR__XMKNOD     ||
 	    sb_nr == SB_NR___XMKNOD    ||
 	    sb_nr == SB_NR___XMKNODAT  ||
 	    sb_nr == SB_NR_MKFIFO      ||
+	    sb_nr == SB_NR_MKFIFOAT    ||
 	    sb_nr == SB_NR_LINK        ||
+	    sb_nr == SB_NR_LINKAT      ||
 	    sb_nr == SB_NR_SYMLINK     ||
+	    sb_nr == SB_NR_SYMLINKAT   ||
 	    sb_nr == SB_NR_RENAME      ||
+	    sb_nr == SB_NR_RENAMEAT    ||
 	    sb_nr == SB_NR_LUTIMES     ||
 	    sb_nr == SB_NR_UTIMENSAT   ||
 	    sb_nr == SB_NR_UTIME       ||
-- 
1.6.2

